Sources/Common/Service/HttpClient.swift

```diff
@@ -16,14 +16,18 @@ public protocol HttpClient: AutoMockable {
 
 // sourcery: InjectRegister = "HttpClient"
 public class CIOHttpClient: HttpClient {
-    private let session: URLSession
     private let baseUrls: HttpBaseUrls
     private var httpRequestRunner: HttpRequestRunner
     private let jsonAdapter: JsonAdapter
     private var globalDataStore: GlobalDataStore
     private let logger: Logger
     private let retryPolicyTimer: SimpleTimer
     private let retryPolicy: HttpRetryPolicy
+    private let cioApiSession: URLSession // only used to call the CIO API.
+    private let publicSession: URLSession // session used to call servers accessible to the public (such as CDNs)
```

Comment:
This is the guts of the PR. We have 2 separate `URLSession` objects that get used at runtime according to the URL host being called.